### ISSUE TITLE
riscv64 test segfault with GCC 14

### ISSUE THREAD

**Issue by marxin** [rui314/mold#1222]
The following failures happen with the latest compiler which might be the culprit of the troubles. All binaries segfault:

```
[ 5403s] The following tests FAILED:
[ 5403s] 	115 - riscv64-ifunc-static-pie (Failed)
[ 5403s] 	202 - riscv64-section-order (Failed)
[ 5403s] 	214 - riscv64-static-pie (Failed)
```

Binaries: 
[riscv64.tar.gz](https://github.com/rui314/mold/files/14662794/riscv64.tar.gz)

**Comment by rui314** [comment]
Could you share the entire test directory?

**Comment by marxin** [comment]
Note I don't have a direct access to the machine, but I need to use `xxd` to dump the file contents. Please list me files you are interested in?

**Comment by marxin** [comment]
@rui314 ping

**Comment by rui314** [comment]
The above change may fix the issue. At least it seems related.

**Comment by marxin** [comment]
Apparently, it hasn't helped with the mentioned commit: https://build.opensuse.org/package/live_build_log/devel:gcc:next/mold/openSUSE_Factory_RISCV/riscv64.
Maybe @andreas-schwab can help you with that?

**Comment by andreas-schwab** [comment]
See dd1a2519

**Comment by marxin** [comment]
Ok, so it's addressed with #1206, am I correct? Can you please address also the other tests mentioned in this issue?
